project/ClientMimaBuild.scala

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.typesafe.tools.mima.core._
+import com.typesafe.tools.mima.plugin.MimaKeys.{mimaBinaryIssueFilters, mimaFailOnNoPrevious, mimaPreviousArtifacts}
+import java.io.FileNotFoundException
+import java.net.URLClassLoader
+import java.util.regex.Pattern
+import sbt._
+import sbt.Keys.{baseDirectory, version}
+import sbt.internal.inc.ScalaInstance
+import scala.io.Source
+
+// Used for spark connect client API compatibility check with sql API.
+// previous artifact: spark-sql.jar
+// current artifact: spark-connect-client-jvm.jar
+object ClientMimaBuild {
+  def mimaSettings: Seq[Setting[_]] = {
+
+    val organization = "org.apache.spark"
+    val preId = "spark-sql"
+
+    Seq(
+      mimaFailOnNoPrevious := true,
+      mimaPreviousArtifacts := Set(organization %% preId % version.value),
+      mimaBinaryIssueFilters ++= Seq(IncludeAny( Seq(
+        IncludeByName("org.apache.spark.sql.Dataset"),
+        IncludeByName("org.apache.spark.sql.SparkSession"),
+        IncludeByName("org.apache.spark.sql.DataFrame"),
+        IncludeByName("org.apache.spark.sql.Column"),
+
+      ) ++ includeMethods(new File(baseDirectory.value, "resources/api.txt"))
+      )) ++ notSupported()
+    )
+  }
+
+  // We start with include filters for as the supported methods is limited.
+  // When the API is stable, we should switch to exclude filters.
+  private def includeMethods(file: File): Seq[ProblemFilter] = {
+    if (!file.exists()) {
+      throw new FileNotFoundException(file.getAbsolutePath)
+    }
+    val source = Source.fromFile(file)
+    source.getLines().map(method => {
+      IncludeByName(method)
+    }).toSeq
+  }
+
+  // I cannot use classLoader inside SBT?
+  // Use a class loader to automatically include the implemented methods in the MiMa check
+  private def includeMethods(
```

Comment:
The classloader cannot find the classes in the jar file. Verified the jar contains the classes. Tried to load as class or load as resources, always returned empty.